pandas/core/arrays/datetimelike.py

```diff
@@ -1495,7 +1495,15 @@ def __rsub__(self, other):
             return (-self) + other
 
         # We get here with e.g. datetime objects
-        return -(self - other)
+        from pandas.core.arrays import DatetimeArray
+
+        datetime_result = self - other
+        if isinstance(datetime_result, DatetimeArray):
```

Comment:
I'm a little nervous enforcing a particular type out of a `rsub` operation.

Could we just catch the `TypeError` and reraise it with a better message?

Reply:
Gotcha, I agree that it would be a better implementation